index_api-v2.yaml

```diff
@@ -0,0 +1,722 @@
+                          - example_schema-v1
+                          - another_example-v1
+                NoResults:
+                  value:
+                    message: "No items found matching given criteria."
+        # TODO: Add non-generic error responses (e.g., `Invalid lat (must be number): {lat}`, etc.)
+        400:
+          $ref: "#/components/responses/BadRequest"
+        # TODO: Add authentication/authorization errors when JWT is introduced
+        # 401:
+        #   $ref: '#/components/responses/Unauthorized'
+        # 403:
+        #   $ref: '#/components/responses/Forbidden'
+        429:
+          $ref: "#/components/responses/TooManyRequests"
+        500:
+          $ref: "#/components/responses/ServerError"
+      # TODO: Add JWT authentication
+      # security:
+      #   - jwt: []
+  /nodes-sync:
+    post:
+      tags:
+        - Node Endpoints
+      summary: Add a node to the index with an immediate status response
+      description: |
+        The same as the `POST /nodes` endpoint but with an immediate success or failure status response
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/PostNodeSyncRequest"
+            example:
+              profile_url: "https://somenode.org/optional-subdirectory/node-profile.json"
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  data:
+                    $ref: "#/components/schemas/PostNodeSyncResponse"
+              examples:
+                received:
+                  value:
+                    data:
+                      node_id: "a55964aeaae9625dc2b8dbdb1c4ce0ed1e658483f44cf2be1a6479fe5e144d38"
+                      profile_url: "https://somenode.org/optional-subdirectory/node-profile.json"
+                      status: "received"
+                validated:
+                  value:
+                    data:
+                      node_id: "a55964aeaae9625dc2b8dbdb1c4ce0ed1e658483f44cf2be1a6479fe5e144d38"
+                      profile_url: "https://somenode.org/optional-subdirectory/node-profile.json"
+                      status: "validated"
+                      last_updated: 1601979232403
+                      profile_hash: "c24d14c2c75f55d334a7e0ccf4d35a063a2582a7abb91e16d326f6613b9602bf"
+                posted:
+                  value:
+                    data:
+                      node_id: "a55964aeaae9625dc2b8dbdb1c4ce0ed1e658483f44cf2be1a6479fe5e144d38"
+                      profile_url: "https://somenode.org/optional-subdirectory/node-profile.json"
+                      status: "posted"
+                      last_updated: 1601979232403
+                      profile_hash: "c24d14c2c75f55d334a7e0ccf4d35a063a2582a7abb91e16d326f6613b9602bf"
+                deleted:
```

Comment:
This probably won't happen, no?

Reply:
What won't happen? Please elaborate.

Reply:
Sorry -- that was rather cryptic! Is there a scenario in which the POST /nodes-sync endpoint would return a status `deleted`?

Reply:
Good point. It probably should not. But there might be an edge case if the content of the profile_url changes from valid to invalid JSON and then the profile_url is sent again to the `/node-sync` endpoint. I need to test this out while making the change to treat invalid JSON responses as 404s.

Reply:
In the end, this won't happen so the `delete` example has been removed.